app/controllers/resources_controller.rb

```diff
@@ -142,18 +140,40 @@ def get_fileset_uuid(external_id)
     external_id.split("#{ID_NAMESPACE}/fileSet/").second.split('-', 2).second
   end
 
-  # rubocop:disable Metrics/AbcSize
-  def decorate_file(file:, version:, external_id: nil)
-    blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
-    file[:externalIdentifier] = external_id if external_id
-    file[:version] = version
+  def metadata_for_blob(blob, file)
+    file.delete(:externalIdentifier)
     file[:size] = blob.byte_size
     file[:hasMimeType] = blob.content_type || 'application/octet-stream'
     declared_md5 = file[:hasMessageDigests].find { |digest| digest.fetch(:type) == 'md5' }.fetch(:digest)
     calculated_md5 = base64_to_hexdigest(blob.checksum)
     raise BlobError, "MD5 mismatch for #{file[:filename]}" if declared_md5 != calculated_md5
   end
-  # rubocop:enable Metrics/AbcSize
+
+  def metadata_for_file(globus_file, file)
+    raise GlobusNotFoundError, "Globus file [#{globus_file}] not found." unless File.exist?(globus_file)
+
+    file[:size] = File.size(globus_file)
+    file[:hasMessageDigests] = [
+      { type: 'md5', digest: Digest::MD5.file(globus_file).hexdigest },
+      { type: 'sha1', digest: Digest::SHA1.file(globus_file).hexdigest }
+    ]
+    file[:hasMimeType] = Marcel::MimeType.for Pathname.new(globus_file)
+  end
+
+  def decorate_file(file:, version:, external_id: nil)
+    if signed_id?(file[:externalIdentifier])
+      blob = blob_for_signed_id(file.delete(:externalIdentifier), file[:filename])
+      metadata_for_blob(blob, file)
+    else
```

Comment:
Should we consider an `elsif` here that checks if the ID looks like a Globus URI? The removal of the check up above means that the use case of "edit a cocina that already has files" will wind up being considered as a globus thing, I think?